src/vsr/replica.zig

```diff
@@ -466,6 +454,38 @@ pub fn ReplicaType(
                 }
             }
 
+            // If we were a lagging backup that installed an SV but didn't finish fast-forwarding,
+            // the vsr_headers head op may be part of the checkpoint after this one.
+            maybe(vsr_headers.slice[0].op > self.op_checkpoint_trigger());
+
+            var op_head = for (vsr_headers.slice) |*header| {
+                if (header.op <= self.op_checkpoint_trigger()) break header.op;
+            } else unreachable;
+            assert(op_head <= self.op_checkpoint_trigger());
+
+            for (self.journal.headers) |*header| {
+                assert(header.op <= self.op_checkpoint_trigger());
+                if (header.command == .prepare and header.op > op_head) {
+                    assert(self.log_view >= header.view);
+                    // Typically if there is an op in the WAL higher than the durable headers'
+                    // head op, we must have crashed with a durable SV, not a durable DVC.
+                    //
+                    // However, we could have joined a view and finished some repair before ever
+                    // updating our view_durable.
+                    //
+                    // To avoid special-casing this all over, we pretend this higher op doesn't
+                    // exist. This is safe because the prior view-change didn't complete.
+                    maybe(self.log_view == self.view);
+
+                    if (self.log_view == self.view) op_head = header.op;
+                }
+            }
+            self.op = op_head;
```

Comment:
I wonder if `init` should set `.op = undefined`, so that we assert if we touch it before actually initializing.